eav/models.py

```diff
@@ -245,7 +253,9 @@ def validate_value(self, value):
             validator(value)
 
         if self.datatype == self.TYPE_ENUM:
-            if value not in self.enum_group.values.all():
+            if isinstance(value, EnumValue):
+                value = value.value
+            if not self.enum_group.values.filter(value=value).exists():
```

Comment:
I've used `exists()` here which also optimises the code. Iteration via the entire queryset would be much slower then a simple exists query.